src/main/java/com/example/nlearn/services/TestService.java

```diff
@@ -0,0 +1,109 @@
+package com.example.nlearn.services;
+
+import com.example.nlearn.models.FullTest;
+import com.example.nlearn.models.Group;
+import com.example.nlearn.models.Question;
+import com.example.nlearn.models.Test;
+import com.example.nlearn.repos.GroupRepository;
+import com.example.nlearn.repos.QuestionRepository;
+import com.example.nlearn.repos.TestRepository;
+import com.example.nlearn.repos.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.CrossOrigin;
+
+import java.util.List;
+
+@Service
+@CrossOrigin
+public class TestService {
+
+    @Autowired
+    private QuestionRepository questionRepository;
```

Comment:
Service of a domain object should only have access to the repository it belongs to e.g. `TestService` should only have dependency on `TestRepository` and not `GroupRepository` or `UserRepository`. If you want to access some information related to other domain objects inside of this service you should add a method in `UserService` or `GroupService` and use it here as a dependency.